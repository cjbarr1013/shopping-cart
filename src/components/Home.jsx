import { useNavigate } from 'react-router';
import styles from '../styles/Home.module.css';
import Button from './Button';

function Home() {
  const navigate = useNavigate();

  return (
    <div className={styles.container}>
      <div className={styles.content}>
        <h1>The Store</h1>
        <Button
          text="Shop Now"
          handleClick={() => navigate('/store/shop')}
          type="primary"
        ></Button>
      </div>
    </div>
  );
}

export default Home;
