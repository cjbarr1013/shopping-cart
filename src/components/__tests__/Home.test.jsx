import { vi, describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Home from '../Home';

// Mock the useNavigate function globally
const mockNavigate = vi.fn();
vi.mock('react-router', async () => {
  const actual = await vi.importActual('react-router');
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});

// Clear mock calls before each test
beforeEach(() => {
  mockNavigate.mockClear();
});

describe('Home Component', () => {
  it('renders the welcome message', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    // Check if the welcome message is rendered
    expect(
      screen.getByRole('heading', { name: /welcome to/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /the store/i })
    ).toBeInTheDocument();
  });

  it('renders the SHOP NOW button', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    // Check if the button is rendered
    const button = screen.getByRole('button', { name: /shop now/i });
    expect(button).toBeInTheDocument();
  });

  it('navigates to the shop page when SHOP NOW button is clicked', async () => {
    const user = userEvent.setup();

    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    // Simulate button click
    const button = screen.getByRole('button', { name: /shop now/i });
    await user.click(button);

    // Check if navigation was triggered
    expect(mockNavigate).toHaveBeenCalledWith('/store/shop');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });
});
