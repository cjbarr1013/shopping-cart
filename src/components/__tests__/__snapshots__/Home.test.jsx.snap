// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_b140de"
  >
    <div
      class="_content_b140de"
    >
      <h1>
        <span>
          Welcome to
        </span>
        <span>
          THE STORE
        </span>
      </h1>
      <button
        class="_primary_4048ec"
      >
        SHOP NOW
      </button>
    </div>
  </div>
</DocumentFragment>
`;
